src/typescript/azure-openapi-validator/rules/PathResourceProviderNamePascalCase.ts

```diff
@@ -0,0 +1,41 @@
+import { MergeStates, OpenApiTypes } from "./../rule";
+import { rules } from "../rule";
+import { getAllResourceProvidersFromPath } from "../rules/utilities/rules-helper";
+
+export const PathResourceProviderNamePascalCase: string =
+  "PathResourceProviderNamePascalCase";
+
+rules.push({
+  id: "R3020",
+  name: PathResourceProviderNamePascalCase,
+  severity: "warning",
+  category: "ARMViolation",
+  mergeState: MergeStates.individual,
+  openapiType: OpenApiTypes.arm,
+
+  run: function*(doc, node, path) {
+    if (node.paths !== undefined) {
+      const msg: string =
+        "Resource provider naming must follow the pascal case.";
+      const paths: string[] = Object.keys(node.paths);
+      for (const it of paths) {
+        const resourceProviders = getAllResourceProvidersFromPath(it);
+
+        if (resourceProviders.some(rp => !resourceProviderMustPascalCase(rp))) {
+          yield {
+            message: `${msg} Path: '${it}'`,
+            location: path.concat(["paths"])
+          };
+        }
+      }
+    }
+  }
+});
+
+function resourceProviderMustPascalCase(resourceProvider: string): boolean {
```

Comment:
Could you move this function into a common place?

Reply:
Good idea. Added some unittest to cover.